java/CMakeLists.txt

```diff
@@ -54,4 +69,72 @@ if(JNI_FOUND)
       LIBRARY DESTINATION ${JAVA_INSTALL_PATH}
     )
   endif()
-endif()
+
+  if(NOT WIN32)
+    # Bare Library targed to use for spark
+    find_package(Java)
+    include(UseJava)
+
+    # Ensure target directory exists
+    file(MAKE_DIRECTORY target/classes)
+    file(MAKE_DIRECTORY target/test-classes)
+    file(MAKE_DIRECTORY target/bin/natives/linux_64)
+
+    # required for jni header generation
+    # add_custom_target(javacompile 
```

Comment:
Are these commented out sections here needed?

Reply:
to follow the old jni and make it less of a hassle I checked in the generated JNI headers. It's helpful during dev to have those around.

Reply:
added development comment